### DISCUSSION TITLE
TypeError: directiveValue is not iterable

### DISCUSSION THREAD

**Discussion by lefrancois** [directus/directus#13556] [Archive]
### Describe the Bug

After Updating to version 9.11.1 on an Ubuntu server the following error message appears while starting the server.
Everything worked fine until this update (last update was about 4 week ago).

### To Reproduce

`npm update`
`npx directus database migrate:latest`
`npm run start`

### Errors Shown

```
> impel@1.0.0 start
> directus start

TypeError: directiveValue is not iterable
    at normalizeDirectives (/var/www/directus/impel/node_modules/helmet/dist/middlewares/content-security-policy/index.js:64:25)
    at Function.contentSecurityPolicy (/var/www/directus/impel/node_modules/helmet/dist/middlewares/content-security-policy/index.js:117:31)
    at createApp (/var/www/directus/impel/node_modules/directus/dist/app.js:104:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async createServer (/var/www/directus/impel/node_modules/directus/dist/server.js:41:38)
    at async Command.startServer (/var/www/directus/impel/node_modules/directus/dist/server.js:128:20)
    at async Command.parseAsync (/var/www/directus/impel/node_modules/commander/lib/command.js:923:5)
```

### What version of Directus are you using?

9.11.1

### What version of Node.js are you using?

16.13.1

### What database are you using?

MySql 8

### What browser are you using?

n/a

### How are you deploying Directus?

locally

**Comment by azrikahar** [comment]
Are you using any `CONTENT_SECURITY_POLICY` (or `ASSETS_CONTENT_SECURITY_POLICY`) environment variables? Would you mind sharing them here? Feel free to redact things like domain names into placeholders like `example.com`.

I believe it's one of the more likely cause of such error, as it shouldn't, or at least so for not reported yet, to be something to do with the out-of-the-box or default config.

**Reply by lefrancois** [reply]
Hey @azrikahar,

thanks for your reply and yes, after removing this part in the .env everything works du e again:

```
CONTENT_SECURITY_POLICY_DIRECTIVES__UPGRADE_INSECURE_REQUESTS=true
```

**Reply by azrikahar** [reply]
Ah that makes sense. Based on [Helmet's docs](https://helmetjs.github.io/) (which is the one using these CSP configs), it should be an empty array instead as a signal for _enabling_ it, so you should do the following instead:

```
CONTENT_SECURITY_POLICY_DIRECTIVES__UPGRADE_INSECURE_REQUESTS="[]"
```

tested locally and `true` does cause the same issue, and `[]` solves it for me!

That said, it's rather odd how you _didn't_ get this error, since a fairly similar issue in their repo was almost 2 year ago: [helmetjs/helmet/issues/321](https://github.com/helmetjs/helmet/issues/242#issuecomment-670238460), but I can only suspect that either your setup was never actually using it prior the upgrade, or it's somehow "fixed" (since `true` was never valid) by the few environment-variables-related changes in recent Directus versions.

**Reply by lefrancois** [reply]
Hey @azrikahar,

thanks for the hint, working fine here too! :)
